src/sdl/SDLFactory.ts

```diff
@@ -0,0 +1,23 @@
+import { NetworkId } from "../types/network";
+import { NetworkVersion, BetaSdl, NetworkBeta2, NetworkBeta3 } from "./types";
+import { SDL } from "./index";
+import { YamlSDLParser } from "./YamlSDLParser";
+
+export class SDLFactory {
```

Comment:
Having a dedicated factory rather then the one in the target class is a better practice in terms of separation of concerns and as a result in maintainability and testability